src/test/ui/consts/rustc-const-stability-require-const.rs

```diff
@@ -0,0 +1,52 @@
+#![crate_type = "lib"]
+#![feature(staged_api)]
+#![stable(feature = "foo", since = "1.0.0")]
+
+#[stable(feature = "foo", since = "1.0.0")]
+#[rustc_const_unstable(feature = "const_foo", issue = "none")]
+pub fn foo() {}
+//~^ ERROR attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function or method to be `const`
+
+#[stable(feature = "bar", since = "1.0.0")]
+#[rustc_const_stable(feature = "const_bar", since = "1.0.0")]
+pub fn bar() {}
+//~^ ERROR attributes `#[rustc_const_unstable]` and `#[rustc_const_stable]` require the function or method to be `const`
+
+#[stable(feature = "potato", since = "1.0.0")]
+pub struct Potato;
+
+impl Potato {
```

Comment:
please also add a test for the unstable `impl const Trait for Type`, the attribute works on the impl there, not on the methods. the method sigs also have no const modifier

Reply:
I think that's what is being tested on this one, no?

https://github.com/rust-lang/rust/pull/86838/files#diff-72c09c5a5824317fb4e9db25b7258c43af584148ab17b1d2eb0f9639894809c5

Reply:
apologies, I completely overlooked that

Reply:
No need to apologies! Thank you for giving me some of your time to check up on that MR!

On a side note, I'd like to get more involved as I've been a very happy user for a while. Any part of the compiler/libraries I should look at to be mostly useful?